In [2]:
#import pandas and os
import pandas as pd
import os

## Merging 12 months of data in single file

In [3]:
# test if data url works
# df = pd.read_csv("Sales_Data/Sales_April_2019.csv")
# df.head()

In [4]:
files = [file for file in os.listdir('Sales_Data')]

all_months_data =  pd.DataFrame()

for file in files:
    if file != ".DS_Store":
#         print(file)
        df = pd.read_csv("Sales_Data/"+file)
        all_months_data = pd.concat([all_months_data, df])

In [5]:
all_months_data.to_csv("all_data.csv")

In [6]:
all_data = pd.read_csv("all_data.csv")
all_data.shape

(186850, 7)

## Data Cleanup

### Find all NaN

In [7]:
row_has_NaN = all_data.isnull().any(axis=1)
rows_with_NaN = all_data[row_has_NaN]
rows_with_NaN.shape
all_data = all_data.dropna(how='any')
all_data.head(2)

Unnamed: 0 Order ID             Product Quantity Ordered Price Each  \
0           0   295665  Macbook Pro Laptop                1       1700   
1           1   295666  LG Washing Machine                1      600.0   

       Order Date                        Purchase Address  
0  12/30/19 00:01  136 Church St, New York City, NY 10001  
1  12/29/19 07:03     562 2nd St, New York City, NY 10001

*** Found 545 row out of all_data containing NaN and removed ***

In [15]:
#Identify and Remove Rows containing "Or"
rows_with_or = all_data[all_data['Order Date'].str[0:2] == "Or"]
rows_with_or.shape

(355, 8)

*** Found 355 rows out of all_data containing string Or in the column - Order Data***

In [17]:
all_data = all_data[all_data['Order Date'].str[0:2] != "Or"]
all_data.shape

(185950, 8)

In [21]:
#convert string of columns
all_data['Quantity Ordered'] = pd.to_numeric(all_data['Quantity Ordered'])
all_data['Price Each'] = pd.to_numeric(all_data['Price Each'])

### Identify the type of all columns

In [8]:
features = all_data.columns
type(features)

pandas.core.indexes.base.Index

In [9]:
for feature in features:
    type_of_feature = type(feature)
    print(feature,"->", type_of_feature)

Unnamed: 0 -> <class 'str'>
Order ID -> <class 'str'>
Product -> <class 'str'>
Quantity Ordered -> <class 'str'>
Price Each -> <class 'str'>
Order Date -> <class 'str'>
Purchase Address -> <class 'str'>


### What was the best month for sales? How much was earned that month?

- Add column to seperate the month
- Add column for Daily Sales Value

In [19]:
all_data['Month'] = all_data['Order Date'].str[0:2]
all_data['Month'] = pd.to_numeric(all_data['Month'])
# all_data.head(2)
all_data['Month'].unique()

array([12,  1,  4,  5,  2,  3,  8,  9,  6, 11, 10,  7])

In [22]:
all_data["Sales"] = all_data['Quantity Ordered'] * all_data['Price Each']

In [24]:
all_data.tail()

Unnamed: 0 Order ID                 Product  Quantity Ordered  \
186845       13617   222905  AAA Batteries (4-pack)                 1   
186846       13618   222906        27in FHD Monitor                 1   
186847       13619   222907    USB-C Charging Cable                 1   
186848       13620   222908    USB-C Charging Cable                 1   
186849       13621   222909  AAA Batteries (4-pack)                 1   

        Price Each      Order Date                       Purchase Address  \
186845        2.99  06/07/19 19:02          795 Pine St, Boston, MA 02215   
186846      149.99  06/01/19 19:29  495 North St, New York City, NY 10001   
186847       11.95  06/22/19 18:57  319 Ridge St, San Francisco, CA 94016   
186848       11.95  06/26/19 18:35   916 Main St, San Francisco, CA 94016   
186849        2.99  06/25/19 14:33         209 11th St, Atlanta, GA 30301   

        Month   Sales  
186845      6    2.99  
186846      6  149.99  
186847      6   11.95  
186848      6   11.95  
186849      6    2.99